gadgets/mri_core/CombineGadget.cpp

```diff
@@ -1,68 +1,70 @@
 #include "CombineGadget.h"
 
-namespace Gadgetron{
+using namespace Gadgetron;
+using namespace Gadgetron::Core;
 
-  CombineGadget::CombineGadget() {}
-  CombineGadget::~CombineGadget() {}
+namespace {
+  template <class T> Image<T> combine(Image<T>& image) {
+      GDEBUG("CombineGadget is not well defined for real-valued images. Doing nothing.");
+      return image;
+  }
 
-int CombineGadget::
-process( GadgetContainerMessage<ISMRMRD::ImageHeader>* m1,
-	 GadgetContainerMessage< hoNDArray< std::complex<float> > >* m2)
-{
+  template <class T>
+  Image<std::complex<T>> combine(Image<std::complex<T>>& image) {
+    auto& header = std::get<ISMRMRD::ImageHeader>(image);
+      auto& data = std::get<hoNDArray<std::complex<T>>>(image);
 
-  // Get the dimensions
-  size_t nx = m2->getObjectPtr()->get_size(0);
-  size_t ny = m2->getObjectPtr()->get_size(1);
-  size_t nz = m2->getObjectPtr()->get_size(2);
-  size_t nc = m2->getObjectPtr()->get_size(3);
+      // Get the dimensions
+      size_t nx = data.get_size(0);
+      size_t ny = data.get_size(1);
+      size_t nz = data.get_size(2);
+      size_t nc = data.get_size(3);
 
-  // Create a new message with an hoNDArray for the combined image
-  GadgetContainerMessage< hoNDArray<std::complex<float> > >* m3 = 
-    new GadgetContainerMessage< hoNDArray<std::complex<float> > >();
+      // Create a new message with an hoNDArray for the combined image
+      hoNDArray<std::complex<T>> m3 = hoNDArray<std::complex<T>>();
```

Comment:
I don't think we should continue with the `m1`, `m2`, etc. variable names. It is legacy from when they are chained messages.

Reply:
Yup - I left those in place since I was trying to reduce the amount of lines that were changing and make it more clear that logic was largely untouched for review, but I like changing things to a more self-commenting style. Will update.